_tools/eslint-rules/no-assign-mutated-array.mjs

```diff
@@ -0,0 +1,233 @@
+/**
+ **Derivative work of `no-assign-mutated-array.ts` by Nicholas Jamieson.
+ * @file Ban direct assignment of arrays mutated by impure array mehods.
```

Comment:
I don't think this is necessary. This isn't a convention in this codebase.

Reply:
The point of having this lint rule here is to serve as an example of what having a codebase is like that has custom rules in 2020. The rule itself is still a draft, please be patient, but it lints for [a common pitfall in JavaScript](https://twitter.com/_jayphelps/status/1196249744902967296) and if I could provide any codebase with one custom rule it would probably be this one. The unique thing about [this rule](https://twitter.com/ncjamieson/status/1196253275831406593) is that it's an ES Module and leverages the TypeScript API (extremely rare). This pattern is very new, which is why it's still unpolished, but I ask that you please have a little bit of patience. It's almost done!

Reply:
given that we have [no uses of sort anywhere in this repository](https://github.com/tc39/tc39.github.io/search?q=sort), this rule is not warranted. There is no reason to add rules like this, as they are just noise. Please only add things as necessary to the codebase, otherwise it will be confusing for new maintainers. 

Additionally, in the case we need a new plugin: if an existing, maintained plugin exists, please use that instead of writing one that we would need to maintain. We don't need anything new or experimental for this. It is just linting.

This change makes no sense here. I am getting frustrated that my comments are being ignore and argued with. This has happened multiple times. I don't have the time to fight about this. I will simply start closing and reverting pull requests if they are not addressed. I hope that is clear: my time is limited. I will not spend it arguing.

Reply:
Sorry, I didn't mean to come off as sounding argumentative. You are right though, if you want to keep things in this repo as simple as possible, then custom rules will have to come from elsewhere. Either way, your feedback has been incorporated.